lib/pkcs7.js

```diff
@@ -437,22 +459,28 @@ p7.createSignedData = function() {
   }
 
   function addSignerInfos(mds) {
+    var content;
+
+    if (msg.detachedContent) {
+      content = msg.detachedContent;
+    } else {
+      // get ContentInfo content
+      content = msg.contentInfo.value[1];
+      // skip [0] EXPLICIT content wrapper
+      content = content.value[0];
+    }
+
     // Note: ContentInfo is a SEQUENCE with 2 values, second value is
     // the content field and is optional for a ContentInfo but required here
     // since signers are present
-    if(msg.contentInfo.value.length < 2) {
+    if(!content) {
```

Comment:
since `contentInfo` is no longer being used here, the above comment becomes irrelevant.  Can it be moved to some other useful location?

Reply:
Fixed in https://github.com/digitalbazaar/forge/pull/605/commits/09afde45c47fa21f58038bf1685ff06b9410ceb2